Mathlib/Algebra/BigOperators/Balance.lean

```diff
@@ -0,0 +1,59 @@
+/-
+Copyright (c) 2024 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import Mathlib.Algebra.BigOperators.Expect
+
+/-!
+# Balancing a function
+
+This file defines the balancing of a function `f`, defined as `f` minus its average.
+
+This is the unique function `g` such that `f a - f b = g a - g b` for all `a` and `b`, and
+`∑ a, g a = 0`. This is particularly useful in Fourier analysis as `f` and `g` then have the same
+Fourier transform, except in the `0`-th frequency where the Fourier transform of `g` vanishes.
+-/
+
+open Function
+open Fintype (card)
+open scoped BigOperators Pointwise NNRat
+
+variable {ι F G K : Type*}
+
+namespace Finset
+
+section AddCommGroup
+variable [AddCommGroup G] [Module ℚ≥0 G] [Field K] [Module ℚ≥0 K] {s : Finset ι}
```

Comment:
```suggestion
variable [AddCommGroup G] [Module ℚ≥0 G] [AddCommGroup K] [Module ℚ≥0 K] 
```
I didn't see why it was relevant that `K` should be a field. Maybe rename it to another letter if it's indeed not needed?

Reply:
This was a remnant from before I generalised from fields to `NNRat`-modules. Well spotted!